src/legacy/ui/public/vis/agg_configs.ts

```diff
@@ -134,45 +144,45 @@ class AggConfigs extends IndexedArray {
    * Ignores the non-array indexes
    * @param aggConfigs an AggConfigs instance
    */
-  jsonDataEquals(aggConfigs) {
-    if (aggConfigs.length !== this.length) {
+  jsonDataEquals(aggConfigs: AggConfig[]) {
+    if (aggConfigs.length !== this.aggs.length) {
       return false;
     }
-    for (let i = 0; i < this.length; i += 1) {
-      if (!_.isEqual(aggConfigs[i].toJSON(), this[i].toJSON())) {
+    for (let i = 0; i < this.aggs.length; i += 1) {
+      if (!_.isEqual(aggConfigs[i].toJSON(), this.aggs[i].toJSON())) {
         return false;
       }
     }
     return true;
   }
 
-  toDsl(hierarchical = false) {
+  toDsl(hierarchical: boolean = false) {
     const dslTopLvl = {};
-    let dslLvlCursor;
-    let nestedMetrics;
+    let dslLvlCursor: Record<string, any>;
+    let nestedMetrics: unknown[];
 
     if (hierarchical) {
       // collect all metrics, and filter out the ones that we won't be copying
-      nestedMetrics = _(this.byTypeType.metrics)
-        .filter(function (agg) {
-          return agg.type.name !== 'count';
+      nestedMetrics = _(this.aggs)
```

Comment:
nit: This is just doing a `.filter` and `.map`... lodash probably isn't necessary at all here. (You could even replace with a single reduce)